src/core/heila-core.js

```diff
@@ -0,0 +1,311 @@
+    .then(rows => {
+      if (_.isEmpty(rows)) {
+        throw new Error('Heila row creation failed: ' + dbRow);
+      }
+      if (heila.pushToken) {
+        // this adds the pushToken to Firebase database
+        // so that the "send push message to user" function
+        // triggered by new chat msg write can use the pushToken immediately
+        functionCore.addPushNotificationTokenForUserId(heila.userId, heila.pushToken);
+      }
+      return rows.length;
+    });
+}
+
+function updateHeila(heila) {
+  const dbRow = _makeHeilaDbRow(heila);
+  return knex('heilas').returning('userId').update(dbRow)
+    .where('uuid', heila.uuid)
+    .then(rows => {
+      if (_.isEmpty(rows)) {
+        throw new Error('Heila row update failed: ' + dbRow);
+      }
+
+      if (heila.pushToken) {
+        // this adds the pushToken to Firebase database
+        // so that the "send push message to user" function
+        // triggered by new chat msg write can use the pushToken immediately
+        functionCore.addPushNotificationTokenForUserId(heila.userId, heila.pushToken);
+      }
+
+      return rows.length;
+    })
+    .catch(err => {
+      console.log('something went wrong when updating heila profile');
+      console.log(err);
+      return -1;
+    });
+}
+
+function _findUserIdByUuid(uuid) {
+  return knex('users')
+    .select('users.*')
+    .where({ uuid: uuid })
+    .then(rows => {
+      return rows[0].id;
+    })
+}
+
+function findByUuid(uuid) {
+  console.log('heila-core::findByUuid ' + uuid);
+  return knex('heilas')
+    .select(
+      'heilas.*'
+    )
+    .where({ uuid: uuid })
+    .then(rows => {
+      if (_.isEmpty(rows)) {
+        return null;
+      }
+
+      return _heilaRowToObject(rows[0]);
+    });
+}
+
+function getAllHeilas(uuid) {
+  console.log('getAllHeilas for uuid ' + uuid);
+
+  // palauttaa listan, jossa tällaisia objekteja:
+  // { id: string, name: string, image_url: string,
+  //   team_id: int, bio_text: string, bio_looking_for_type_id: int }
+
+  return _findUserIdByUuid(uuid)
+  .then(userId => {
+
+    return knex('users')
+      .select('users.*')
+      .where({ heila: true })
+      .then(userRows => {
+        if (_.isEmpty(userRows)) {
+          return [];
+        }
+        const heilaIds = userRows.map(user => {
+          return user.uuid;
+        })
+        //console.log(heilaIds)
+        return knex('heilas')
+          .select('heilas.*')
+          .whereIn('uuid', heilaIds)
+          .then(heilaRows => {
+            //console.log(heilaRows);
+            if (_.isEmpty(heilaRows)) {
+              return [];
+            }
+            const unfilteredHeilalist = _heilaRowsToObjectList(_mergeUserHeilaRows(userRows, heilaRows));
+            //console.log('unfilteredHeilaList:')
+            //console.log(unfilteredHeilalist)
+            const myType = unfilteredHeilalist.filter(h => h.id == userId)[0].bio_looking_for_type_id || -1;
+            //console.log('my_type');
```

Comment:
Don't leave commented out console.logs cluttering the code. Debugger is a valuable tool - learning its usage is worth it!

This is a project wide suggestion.